### DISCUSSION TITLE
Does SKElement need to call base.OnRender in OnRender?

### DISCUSSION THREAD

**Discussion by TruePluto** [mono/SkiaSharp#2846] [General]
Does SKElement need to call base.OnRender in OnRender? The parameter of the OnRender method is DrawingContext drawingContext used for drawing, and there is no other drawing logic. SKElement takes over the drawing of the interface, so base.OnRender is not needed.

**Comment by mattleibow** [comment]
Are you asking about this line: https://github.com/mono/SkiaSharp/blob/main/source/SkiaSharp.Views/SkiaSharp.Views.WPF/SKElement.cs#L47

Are you noticing any issues as a result or just curious?

If this code is to be believed, then it is empty and maybe not needed: https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/UIElement.cs#L979-L981

I see other controls in the source do not call base.
